docs/source/task-launch.rst

```diff
@@ -84,71 +105,131 @@ We can submit tasks from other tasks.  This allows us to make decisions while
 on worker nodes.
 
 To submit new tasks from a worker that worker must first create a new client
-object that connects to the scheduler.  There is a convenience function to do
-this for you so that you don't have to pass around connection information.
-However you must use this function ``worker_client`` as a context manager to
-ensure proper cleanup on the worker.
+object that connects to the scheduler. There are three options for this:
+
+1. ``dask.delayed`` and ``dask.compute``
+2. ``get_client`` with ``secede`` and ``rejoin``
+3. ``worker_client``
+
+.. _Scheduling Policies: https://distributed.readthedocs.io/en/latest/scheduling-policies.html
+.. _Scheduling State: https://distributed.readthedocs.io/en/latest/scheduling-state.html
+.. _The compute function: https://dask.pydata.org/en/latest/scheduler-overview.html#the-compute-function
+
+``dask.delayed``
+~~~~~~~~~~~~~~~~
+
+``dask.delayed`` behaves as normal: it submits the functions to the graph,
+optimizes for less bandwidth/computation and gathers the results.
+For more detail, see `dask.delayed`_.
 
 .. code-block:: python
 
-   from distributed import worker_client
+    In [1]: from distributed import Client
+    In [2]: from dask import delayed, compute
 
-   def process_all(data):
-       with worker_client() as e:
-           elements = e.scatter(data)
-           futures = e.map(process_element, elements)
-           analysis = e.submit(aggregate, futures)
-           result = analysis.result()
-       return result
+    In [3]: # these features require the newer dask.distributed scheduler
+    In [4]: client = Client()
 
-    analysis = e.submit(process_all, data)  # spawns many tasks
+    In [4]: @delayed
+       ...: def fib(n):
+       ...:     if n < 2:
+       ...:         return n
+       ...:     # We can use dask.delayed and dask.compute to launch
+       ...:     # computation from within tasks
+       ...:     a = fib(n-1)  # these calls are delayed
+       ...:     b = fib(n-2)
+       ...:     a, b = compute(a, b)  # execute both in parallel
+       ...:     return a + b
 
-This approach is somewhat complex but very powerful.  It allows you to spawn
-tasks that themselves act as potentially long-running clients, managing their
-own independent workloads.
+    In [5]: fib(10).compute()
+    Out[5]: 55
 
-Extended Example
-~~~~~~~~~~~~~~~~
+.. _dask.delayed: https://dask.pydata.org/en/latest/delayed.html
 
-This example computing the Fibonacci numbers creates tasks that submit tasks
-that submit tasks that submit other tasks, etc..
+``get_client``
+~~~~~~~~~~~~~~
+
+``distributed.get_client`` allows individual workers to access the ``client``
+that the scheduler uses:
```

Comment:
This is worded oddly.  The scheduler doesn't use clients.  How about

"The get_client function provides a normal Client object that gives full access to the dask cluster, including the ability to submit, scatter, and gather results"

Ideally we would also link to `get_client` in the internal API (there is some sphinx magic to do this) and drop the `distributed` prefix (I'd like to avoid people referring to this as a separate library).

Reply:
I found a way to link to the `get_client` internal API and remove the `distributed` prefix.